x/bank/proto/cosmos/bank/v2/tx.proto

```diff
@@ -52,14 +79,24 @@ message MsgMint {
   // authority is the address that controls the module (defaults to x/gov unless overwritten).
   string authority = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
 
-  string   to_address                      = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
-  repeated cosmos.base.v1beta1.Coin amount = 3 [
-    (gogoproto.nullable)     = false,
-    (amino.dont_omitempty)   = true,
-    (amino.encoding)         = "legacy_coins",
-    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
-  ];
+  string                   to_address = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  cosmos.base.v1beta1.Coin amount     = 3 [(gogoproto.nullable) = false];
 }
 
-// MsgMint defines the response structure for executing a MsgMint message.
+// MsgMintResponse defines the response structure for executing a MsgMint message.
 message MsgMintResponse {}
+
+// MsgBurn is the Msg/Burn request type.
+message MsgBurn {
+  option (cosmos.msg.v1.signer) = "authority";
+  option (amino.name)           = "cosmos-sdk/x/bank/v2/MsgBurn";
+
+  // authority is the address that controls the module (defaults to x/gov unless overwritten).
+  string authority = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+
+  string                   burn_from_address = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
```

Comment:
I have some concerns about this feature. The original token factory was built for a permissioned chain which is always able to do this via migration code. 
When a contract can use the TF to mint and burn tokens from any address, this opens the door to fraud.
The burn use-case is not new in other chains but what they do instead is providing transparency via metadata that the users know this is possible. See https://developers.stellar.org/docs/learn/encyclopedia/transactions-specialized/clawbacks for example